dds/DCPS/Service_Participant.cpp

```diff
@@ -388,6 +390,10 @@ Service_Participant::get_domain_participant_factory(int &argc,
         }
       }
 
+      if (!parse_env()) {
+        return DDS::DomainParticipantFactory::_nil();
```

Comment:
I know it doesn't fail right now, but If this fails, what does that mean? If fails because we couldn't get environment because the platform doesn't have one, then do we really have to fail to create the `DomainParticipantFactory` because of it?